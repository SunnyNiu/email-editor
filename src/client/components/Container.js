import React from 'react';
import { Grid, Cell } from 'styled-css-grid';
import PropTypes from 'prop-types';
import Sidebar from './Sidebar';
import Content from './Content';
import ContentEditor from './ContentEditor';

const Container = props => {
  const { userId } = props.match.params;
  return (
    <Grid columns="1fr 2fr 2fr" gap="4px">
      <Cell minWidth="200px">
        <Sidebar />
      </Cell>
      <Cell>
        <Content userId={userId} />
      </Cell>
      <Cell>
        <ContentEditor />
      </Cell>
    </Grid>
  );
};

Container.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      userId: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  userId: PropTypes.string.isRequired,
};

export default Container;
