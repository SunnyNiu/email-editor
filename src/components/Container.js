import React from 'react';
import { Grid, Cell } from 'styled-css-grid';
import Sidebar from './Sidebar';
import Content from './Content';
import ContentEditor from './ContentEditor';

const Container = () => (
  <Grid columns="1fr 2fr 2fr" gap="4px">
    <Cell minWidth="200px">
      <Sidebar />
    </Cell>
    <Cell>
      <Content />
    </Cell>
    <Cell>
      <ContentEditor />
    </Cell>
  </Grid>
);

export default Container;
