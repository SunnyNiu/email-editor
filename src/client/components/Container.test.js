import React from 'react';
import { shallow } from 'enzyme';

import Container from './Container';

describe('Container component tests', () => {
  it('Create snapshot', () => {
    const wrapper = shallow(<Container />);
    expect(wrapper).toMatchSnapshot();
  });
});
