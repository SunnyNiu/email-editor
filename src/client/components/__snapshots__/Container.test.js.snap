// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component tests Create snapshot 1`] = `
<Grid__Grid
  columns="1fr 2fr 2fr"
  gap="4px"
>
  <Cell__Cell
    minWidth="200px"
  >
    <Connect(Sidebar) />
  </Cell__Cell>
  <Cell__Cell>
    <Connect(Content) />
  </Cell__Cell>
  <Cell__Cell>
    <ContentEditor />
  </Cell__Cell>
</Grid__Grid>
`;
